src/sentry/integrations/slack/requests/base.py

```diff
@@ -17,7 +17,7 @@
 
 
 def _get_field_id_option(data: Mapping[str, Any], field_name: str) -> str | None:
-    id_option: str | None = data.get(f"{field_name}_id") or data.get(field_name, {}).get("id")
+    id_option: str | None = data.get(f"{field_name}_id") or (data.get(field_name) or {}).get("id")
```

Comment:
Nit: Could be safer to switch over to https://github.com/getsentry/sentry/blob/e715887dce8c7448a857e6f5cbdefab50eeada51/src/sentry/utils/safe.py#L107 here.